syntax = "proto3";

package example;

message Request {

  message Connect {
    enum Mode {
      MODE_AUTO = 0;
      MODE_READ_WRITE = 1;
      MODE_READ_ONLY = 2;
    }

    string file_name = 1;
    Mode mode = 2;
  }
  
  message Query {
    string query = 1;
  }

  oneof message {
    Connect connect = 1;
    Query query = 2;
  }
}

message Response {

  message QueryResult {
    string result = 1;
  }

  QueryResult result = 1;
}
